### PR TITLE
Feat/collapsible datagrid

### PR DESCRIPTION
**Pull request by whispyy**
UK-37

Linked to
https://github.com/ovh-ux/ovh-ui-kit/pull/242
https://github.com/ovh-ux/ovh-ui-kit-documentation/pull/130

### CODE REVIEW

File: packages/oui-datagrid/src/datagrid.controller.js [hunk 1]
```diff
@@ -150,6 +151,17 @@ export default class DatagridController {
...
 
+    toggleSubrows (index) {
+        this.expandedLines[index] = !this.expandedLines[index];
+    }
+
+    hasSubrows (row) {
+        if (!row) {
+            return false;
+        }
+        return this.subrows && row[this.subrows] && row[this.subrows].length;
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.controller.js]: `!_.isEmpty(row[this.subrows])` ?

For a function called `hasSubrows`, it doesn't return a boolean but a number. It seems weird.

File: packages/oui-datagrid/src/datagrid.html [hunk 2]
```diff
@@ -46,10 +46,12 @@
...
                         ng-hide="$ctrl.loading">
-                        <tr class="oui-datagrid__row oui-action-menu-container"
-                            ng-repeat="row in $ctrl.displayedRows track by $index">
+                        <tr class="oui-datagrid__row  oui-action-menu-container"
+                            ng-class="{ 'oui-datagrid__row_expanded': $ctrl.expandedLines[rowIndex] }"
+                            ng-init="rowIndex = $index"
+                            ng-repeat-start="row in $ctrl.displayedRows track by $index">
                             <td class="oui-datagrid__cell"
-                                ng-class="::[column.class]"
+                                ng-class="[column.class, { 'oui-datagrid__cell_expand-button': $index == 0 && $ctrl.hasSubrows(row) }]"
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.html]: ===

File: packages/oui-datagrid/src/datagrid.html [hunk 3]
```diff
@@ -59,6 +61,11 @@
                                     row="row"
                                     column="column">
                                 </oui-datagrid-cell>
+                                <button class="oui-button oui-button_icon-alone oui-datagrid__expand-button"
+                                    ng-if="$index == 0 && $ctrl.hasSubrows(row)"
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.html]: ===

File: packages/oui-datagrid/src/datagrid.directive.js [hunk 4]
```diff
@@ -16,6 +16,7 @@ export default () => {
             rows: "<?",
             rowsLoader: "&?",
             rowLoader: "&?",
+            subrows: "@?",
```
Comment (AxelPeter) [inline packages/oui-datagrid/src/datagrid.directive.js]: No default value for this attribute (eg. `children`) ?
What do you think of `subrows-property` for the attribute name ? it's more explicit and follow the logic of `oui-column`.

**Comment by jleveugle** [comment]
Hello @whispyy,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**